furnace.c

```diff
@@ -106,7 +116,8 @@ tcp_server_close(furnace_context_t* ctx)
     err = tcp_close(ctx->tcp.client_pcb);
 
     if (err != ERR_OK) {
-      DEBUG_printf("close failed %d, calling abort\n", err);
+      if(debug_layers.server)
+        DEBUG_printf("close failed %d, calling abort\n", err);
```

Comment:
Consider making a new log function like:
```c
#include <stdarg.h>

/* Should probably be part of furnace ctx. */
uint8_t log_bits;

enum log_kind {
  /* 
   *  Do we have some guarantee about the layout of the bit fields from debug_layers?
   *  We might want to have some _Static_assert in place to check in compile time
   *  that our assumptions hold true.  
   */
  LOG_KIND_SERVER = 1 << 0,
  LOG_KIND_THERMOCOUPLE = 1 << 1,
};

static void log_stdout(enum log_kind kind, const char *fmt, ...)
{
  if (log_bits & (uint8_t) kind) == 0)
    return;

  va_list args;
  va_start(args, fmt);

  vprintf(fmt, myargs);

  va_end(args);
}

#define log_stdout_server(fmt, ...) log_stdout(LOG_KIND_SERVER, fmt, __VA_ARGS__)
#define log_stdout_thermocouple(fmt, ...) log_stdout(LOG_KIND_THERMOCOUPLE, fmt, __VA_ARGS__)
```

I hope pico-sdk has vprintf in place for us to use.

More info on [va_list](https://en.cppreference.com/w/c/variadic/va_list) and [va_start](https://en.cppreference.com/w/c/variadic/va_start).